### ISSUE TITLE
`qth_survival_time` raising ambiguous truth value error

### ISSUE THREAD

**Issue by daturkel** [CamDavidsonPilon/lifelines#535]
Here's a minimal example to explain the issue:
```
from numpy.random import exponential
from lifelines import utils as survival_utils
from lifelines import KaplanMeierFitter

T_control = exponential(10, size=250)
T_experiment = exponential(20, size=200)

kmf_control = KaplanMeierFitter()
kmf_control.fit(T_control, label='control')

survival_utils.qth_survival_time(.5,kmf_control.survival_function_)
```
This raises error `The truth value of a Series is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().` because `kmf_control.survival_function_.iloc[-1]` (per [here](https://github.com/CamDavidsonPilon/lifelines/blob/master/lifelines/utils/__init__.py#L87)) returns a series, not a value.

If we replace the last line with this, it works:
`survival_utils.qth_survival_time(0.5,kmf_control.survival_function_.iloc[:,0])`

So I think we can fix this by replacing [this line](https://github.com/CamDavidsonPilon/lifelines/blob/master/lifelines/utils/__init__.py#L87) with `if survival_function.iloc[-1,0] > q:`. 

This isn't caught by the test because the test fakes a survival function with `pd.Series(somelist)`, but what we really have here is a one-column dataframe, not a series.

**Comment by CamDavidsonPilon** [comment]
I love this issue because it has everything:
 - reproducible example
 - explanation why current code fails
 - explanation why the current _tests_ are incomplete

This is great - I'll fix this for the upcoming v0.15.0 release.

**Comment by daturkel** [comment]
Thanks for the super quick turnaround here! Happy I could help.
